crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,443 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+/// The strategy used to position this node.
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum PositionType {
+    /// Positioned in relation to its parent, while considering all sibling-nodes that also have the [`PositionType::Relative`] value.
+    #[default]
+    Relative,
+    /// Positioned in relation to its parent, without considering any sibling nodes.
+    Absolute,
+}
+
+/// The offset of a UI node from its base position
+///
+/// Layouting is performed by the Flexbox layout algorithm where the value of [`Offset`] is considered.
+/// To check the final position of a UI element, read its [`Transform`](bevy_transform::components::Transform) component.
+#[derive(
+    Component,
+    Deref,
+    DerefMut,
+    Copy,
+    Clone,
+    PartialEq,
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Offset(pub UiRect);
+
+/// Controls the size of UI nodes
+///
+/// Layouting is performed by the Flexbox layout algorithm where the value of [`SizeConstraints`] is considered.
+/// To check the actual size of a UI element, read its [`Transform`](bevy_transform::components::Transform) component
+#[derive(Component, Copy, Clone, PartialEq, Debug, Default, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct SizeConstraints {
+    /// The minimum extent, which cannot be violated by the layouting algorithm.
+    pub min: Size,
+    /// The suggested extent, which will be used if other constraints can be resolved.
+    pub suggested: Size,
+    /// The maximum extent, which cannot be violated by the layouting algorithm.
+    pub max: Size,
+    /// The expected aspect ratio, computed as width / height.
```

Comment:
There should be a comment explaining how min constraints overide max constraints.

```suggestion
    /// The minimum extent, which cannot be violated by the layouting algorithm.
    ///
    /// `min` constraints overrides the `max` constraints 
    /// i.e. if `max.width` < `min.width`, then `max.width` will be ignored 
    pub min: Size,
    /// The suggested extent, which will be used if other constraints can be resolved.
    pub min: Size,
    /// The suggested extent, which will be used if other constraints can be resolved.
    pub suggested: Size,
    /// The maximum extent, which cannot be violated by the layouting algorithm.
    ///
    /// `min` constraints overrides the `max` constraints 
    /// i.e. if `max.width` < `min.width`, then `max.width` will be ignored 
    pub max: Size,
    /// The expected aspect ratio, computed as width / height.
```